### ISSUE TITLE
statsd pluging: average latency overflows and sends 8589934592 when original value is 0ms

### ISSUE THREAD

**Issue by xo4n** [collectd/collectd#960]
Running collectd-5.4.2 with the following statsd config

```
Interval  10
<Plugin statsd>
        Host "0.0.0.0"
        Port "8126"
        DeleteCounters false
        DeleteTimers   false
        DeleteGauges   false
        DeleteSets     false
        TimerPercentile 90.0
</Plugin>
```

and sending the following metrics

```
while true
do
        echo "collectd.test.data:1|c" | nc -w 1 -u 191.167.0.211 8126
        echo "collectd.test.data:11|g" | nc -w 1 -u 191.167.0.211 8126
        echo "collectd.test.data:0|ms" | nc -w 1 -u 191.167.0.211 8126
done
````
results in the following values generated by statsd and a wrong calculation for average latency as statsd.latency-collectd_test_data-average 8589934592

```
tcpdump -A -i eth0 port 2030 | egrep stats
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on eth0, link-type EN10MB (Ethernet), capture size 65535 bytes

springfield04.statsd.derive-collectd_test_data 0.299982 1425567085
springfield04.statsd.gauge-collectd_test_data 11.000000 1425567085
springfield04.statsd.latency-collectd_test_data-average 8589934592.000000 1425567085
springfield04.statsd.latency-collectd_test_data-percentile-90 0.000000 1425567085
springfield04.statsd.derive-collectd_test_data 0.400024 1425567095
springfield04.statsd.gauge-collectd_test_data 11.000000 1425567095
springfield04.statsd.latency-collectd_test_data-average 8589934592.000000 1425567095
springfield04.statsd.latency-collectd_test_data-percentile-90 0.000000 1425567095
springfield04.statsd.derive-collectd_test_data 0.300000 1425567105
springfield04.statsd.latency-collectd_test_data-average 8589934592.000000 1425567105
springfield04.statsd.latency-collectd_test_data-percentile-90 0.000000 1425567105
```

**Comment by mfournier** [comment]
I confirm this issue, easily reproducible following the above instructions, thanks @xo4n !

I appears this problem is fixed in the master branch (upcoming 5.5 release), but I'm not sure which patch fixed it and should be backported to 5.4.

**Comment by mfournier** [comment]
804ef69e appears to be the fix. Bummer, this should have been in 5.4.2 :angry:. I've cherry-picked the patch (and also 60dffffa1) to the 5.4 bugfix branch.

Sorry about that, but unless you manually apply this patch, you'll have to wait for 5.4.3 or 5.5.0 to see this fixed...
